### ISSUE TITLE
DOTA2 fails to launch 

### ISSUE THREAD

**Issue by eeyrjmr** [flathub/com.valvesoftware.Steam#1235]
**Game information**
DOTA2

**Distribution name and version where applicable**
Gentoo

**Flatpak info**
Flatpak 1.14.5

flatpak --gl-drivers
default
host

**Problem description**
Since roughly 2weeks ago, DOTA2 will fail to launch via Flatpak Steam.  
The output from a terminal shows this when using: **flatpak run com.valvesoftware.Steam**

```
/bin/sh\0-c\0/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun --  '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh' +@panorama_min_comp_layer_dimension 0 -prewarm_panorama\0
chdir "/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/dota 2 beta"
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
pressure-vessel-adverb[1331]: E: Unable to receive --assign-fd source 6: Bad file descriptor
Uploaded AppInterfaceStats to Steam
```
There is nothing in dmesg or the system log to provide additional clues as to what file descriptor isn't being mapped. 

When launch steam directly  **~/.var/app/com.valvesoftware.Steam/data/Steam/steam.sh** DOTA2 launches as expected 

```
/bin/sh\0-c\0/home/jrb/.var/app/com.valvesoftware.Steam/data/Steam/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/jrb/.var/app/com.valvesoftware.Steam/data/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun --  '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh' +@panorama_min_comp_layer_dimension 0 -prewarm_panorama\0
chdir "/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/dota 2 beta"
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
pid 19868 != 19867, skipping destruction (fork without exec?)
pid 19869 != 19867, skipping destruction (fork without exec?)
Loaded /home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libengine2.so, got 0x5638d5b1a2e0
Using breakpad crash handler
[S_API] SteamAPI_Init(): Loaded '/home/jrb/.var/app/com.valvesoftware.Steam/.local/share/Steam/linux64/steamclient.so' OK.
Setting breakpad minidump AppID = 570
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
01/14 14:59:24 Init: Installing breakpad exception handler for appid(570)/version(8631877)/tid(19871)
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561197974302795 [API loaded yes]
SteamInternal_SetMinidumpSteamID:  Setting Steam ID:  76561197974302795
Setting breakpad minidump AppID = 373300
Loaded libSDL3.so.0, got 0x5638d5c359c0
...
```
with the logs carrying on for further game information. 

**Does this issue reproduce with native Steam**
No

**Comment by smcv** [comment]
This is a regression in the `client_beta` branch of "Steam Linux Runtime 2.0 (soldier)" and "Steam Linux Runtime 3.0 (sniper)". I'll try to fix it in the next beta of each one.

As a workaround, go back to the the default (non-beta) branch of each of those compatibility tools.

**Comment by smcv** [comment]
> This is a regression in the `client_beta` branch of "Steam Linux Runtime 2.0 (soldier)" and "Steam Linux Runtime 3.0 (sniper)". I'll try to fix it in the next beta of each one. See [ValveSoftware/steam-runtime#644](https://github.com/ValveSoftware/steam-runtime/issues/644).

The error `pressure-vessel-adverb[...]: E: Unable to receive --assign-fd source 6: Bad file descriptor` is fixed in today's betas (labelled 0.20240125.x in `SteamLinuxRuntime_*/VERSIONS.txt`).

**Comment by smcv** [comment]
> fixed in today's betas (labelled 0.20240125.x)

These betas have now been promoted to stable releases, so I believe there is nothing to be done here. Please could @eeyrjmr or a maintainer close this?
